### PR TITLE
Enable assignment for QTensor in pytorch frontend

### PR DESCRIPTION
**Pull request by jerryzh168**
Stack:
&nbsp;&nbsp;&nbsp;&nbsp;:black_circle:&nbsp; **#19530 Enable assignment for QTensor in pytorch frontend**&nbsp;&nbsp;[:yellow_heart:](https://our.internmc.facebook.com/intern/diff/D15008160/)

Make copy work with QTensor, enable assignment of QTensor in pytorch frontend.

Differential Revision: [D15008160](https://our.internmc.facebook.com/intern/diff/D15008160/)

### CODE REVIEW

File: aten/src/ATen/native/quantized/Copy.cpp [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+#include <ATen/ATen.h>
+#include <ATen/quantized/Quantizer.h>
+#include <ATen/CPUApplyUtils.h>
+
+namespace at { namespace native {
+Tensor& _s_copy__quantized(
+    Tensor& self,
+    const Tensor& src,
+    bool non_blocking) {
+  AT_CHECK(self.scalar_type() == at::kQInt8, "Quantized copy only works with kQInt8 as source Tensor");
```
Comment (li-roy) [inline aten/src/ATen/native/quantized/Copy.cpp]: I think these two error messages are reversed. QInt8 should be the target, not the source.

File: aten/src/ATen/native/native_functions.yaml [hunk 2]
```diff
@@ -446,6 +446,7 @@
   dispatch:
     CPU: _s_copy__cpu
     CUDA: _s_copy__cuda
+    QuantizedCPU: _s_copy__quantized
```
Comment (li-roy) [inline aten/src/ATen/native/native_functions.yaml]: Just wondering, will we have to support copy between two different quantized tensors?

Comment (li-roy) [inline aten/src/ATen/native/native_functions.yaml]: Just wondering, will we have to support copy between two quantized tensors?

File: aten/src/ATen/native/quantized/Copy.cpp [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+  AT_CHECK(src.scalar_type() == at::kFloat, "Quantized copy only works with kFloat as target Tensor");
```
Comment (li-roy) [inline aten/src/ATen/native/quantized/Copy.cpp]: I think these two checks are reversed. QInt8 should be the target tensor.

**Review by li-roy — APPROVED** [review]

File: aten/src/ATen/native/native_functions.yaml (hunk 2 above)
Comment (jerryzh168) [reply aten/src/ATen/native/native_functions.yaml]: I don't think we will support that in this call, we might have other ways to create a QTensor from another QTensor. or provide some other APIs..

File: torch/csrc/autograd/python_variable_indexing.cpp [hunk 4]
```diff
@@ -334,7 +335,12 @@ int THPVariable_setitem(PyObject* self, PyObject* index, PyObject* py_value) {
 
   auto& self_ = reinterpret_cast<THPVariable*>(self)->cdata;
   OptionalDeviceGuard device_guard(device_of(self_));
-  auto value = valueToTensor(self_.dispatch_type(), self_.scalar_type(), py_value);
+  Variable value;
+  if (isQIntType(self_.scalar_type())) {
+    value = valueToTensor(at::globalContext().getVariableType(at::Backend::CPU, at::kFloat), at::kFloat, py_value);
+  } else {
+    value = valueToTensor(self_.dispatch_type(), self_.scalar_type(), py_value);
```
Comment (jerryzh168) [inline torch/csrc/autograd/python_variable_indexing.cpp]: @li-roy will this affect perf?

Comment (li-roy) [reply torch/csrc/autograd/python_variable_indexing.cpp]: This should be fine

**Comment by ezyang** [comment]
This PR fails lint.

**Pull request closed without merging**